lib/overcommit/git_repo.rb

```diff
@@ -54,16 +54,24 @@ def modified_files(options)
         map { |relative_file| File.expand_path(relative_file) }
     end
 
-    # Returns the names of all files that are tracked by git.
+    # Returns the names of files in the given paths that are tracked by git.
     #
+    # @param paths [Array<String>] variable-length list of paths to check
     # @return [Array<String>] list of absolute file paths
-    def all_files
-      `git ls-files`.
+    def list_files(*paths)
+      `git ls-files #{paths.join(' ')}`.
```

Comment:
Looks like `git ls-tree HEAD --name-only <paths>` is what we want. I'll make the change.